xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/AbstractMessageMacro.java

```diff
@@ -73,4 +93,35 @@ public void prepare(MacroBlock macroBlock) throws MacroPreparationException
     {
         this.contentParser.prepareContentWiki(macroBlock);
     }
+
+    @Override
+    public List<Block> execute(BoxMacroParameters parameters, String content, MacroTransformationContext context) 
+        throws MacroExecutionException 
+    {
+        List<Block> boxFoundation = super.execute(parameters, content, context);
+        if (!boxFoundation.isEmpty() && this.iconName != null) {
+            Block defaultBox = boxFoundation.get(0);
+            // For an easier styling, we wrap the content and title together if they are non-empty and visible
+            if (defaultBox.getChildren().size() > 1) {
+                Block boxTextContent = new GroupBlock(defaultBox.getChildren());
+                defaultBox.setChildren(List.of(boxTextContent));
+            }
+            // Enhance the default box with an icon as the first element.
+            Block iconBlock = iconProvider.get(iconName);
+            // Provide an accessible name besides this icon
+            // This is the responsibility of the message macro and not the iconProvider which should only provide
+            // icons without any semantics
+            if (iconBlock.getClass() == ImageBlock.class) {
+                iconBlock.setAttribute("alt", iconPrettyName);
+            } else if (!iconPrettyName.equals("")) {
```

Comment:
`!iconPrettyName.isEmpty()` or `StringUtils.isNotEmpty(iconPrettyName)`

By the way, we generally prefix fields with `this.`.

Reply:
Addressed both of those in https://github.com/xwiki/xwiki-rendering/pull/296/commits/554112649ff574ae69f7821fbf60a49e5faae5fe :+1: